main.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    i = 0
+    a = is_a?(Range) ? to_a : self
+    while i < a.length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    return to_enum :my_each unless block_given?
+
+    a = is_a?(Range) ? to_a : self
+    while i < a.length
+      yield(a[i], i)
+      i += 1
+    end
+    a
+  end
+
+  def my_select
+    return to_enum :my_each unless block_given?
+
+    arr = []
+    my_each { |x| arr.push(x) if yield(x) }
+    arr
+  end
+
+  def my_all?(arg = nil, &prc)
+    if block_given?
+      my_each { |x| return false if prc.call(x) == false }
+    elsif arg.nil? == false
+      my_each { |x| return false if arg != x }
+    elsif arg.is_a?(Regex)
+      my_each { |x| return false if arg.match?(x.to_s) == false }
+    elsif arg.is_a?(Class)
+      my_each { |x| return false if x.is_a?(arg) == false }
+    elsif block_given? == false
+      to_enum
+    else
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &prc)
+    if block_given?
+      my_each { |x| return true if prc.call(x) }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return true if arg.match?(x.to_s) == true }
+    elsif arg.is_a?(Class)
+      my_each { |x| return true if x.is_a?(arg) }
+    elsif arg.nil? == false
+      my_each { |x| return true if arg == x }
+    elsif block_given? == false
+      to_enum
+    else
+      my_each { |x| return true if x }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |x| return false if prc.call(x) }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return false if arg.match?(x.to_s) == true }
+    elsif arg.is_a?(Class)
+      my_each { |x| return false if x.is_a?(arg) }
+    elsif arg.nil? == false
+      my_each { |x| return false if x == arg }
+    else
+      my_each { |x| return false if x }
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] returns true if the block never returns true for all elements

```ruby
[1, 5, 8].my_none? {|number| number > 10 } == [1, 5, 8].none? {|number| number > 10 } => true
```
- Yours is retrieving an error
![image](https://user-images.githubusercontent.com/52209067/81760152-b6fecd00-9483-11ea-90fb-69cef56e6156.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/81760179-c9790680-9483-11ea-9ac5-e7d208dd427a.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end

```
To make this work we have to use block.call(element) instead of yield(element) in my_any?